### PR TITLE
[slang] Adding a package for the Slang shading language

### PR DESCRIPTION
**Pull request by djohansson**
Adding a package for the Slang shading language (https://github.com/shader-slang/slang).
Uses the Github binary releases found at https://github.com/shader-slang/slang/releases

- #### What does your PR fix?
(New package)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
x64-linux, x64-windows, x86-windows
CI baseline updated: Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes, I hope so.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yup!

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@djohansson, Thanks for your pr , are you still work on this pr?

**Comment by JonLiu1993** [comment]
Pinging @djohansson for response. Is work still being done for this PR?

**Comment by djohansson** [comment]
@JonLiu1993 sorry for the late reply. I have not dropped this, as long as there is not any other port for slang avaliable I want to keep working on it.

**Comment by djohansson** [comment]
Note: There are newer versions of slang avaliable, as well as more supported platforms. But I think it makes sense to get this initial port published before adding more platforms and updating the version.

**Comment by djohansson** [comment]
@JonLiu1993 This PR is ready for review now.

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1014,6 +1014,11 @@ signalrclient:arm-uwp=fail
 simbody:arm64-windows=fail
 skia:arm-uwp=fail
 skia:x64-uwp=fail
+slang:arm64-windows=fail
+slang:arm-uwp=fail
+slang:x64-osx=fail
+slang:x64-uwp=fail
+slang:x64-windows-static=fail
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:1021]: @djohansson, If this port doesn't support this triplet, please add [supports](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md#supports) in vcpkg.json and remove this code from ci.baseline.txt.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
This is great, thank you! Can you check/confirm/fix the usage text? Then this is ready to merge!

slang

New port checklist:
* Review the code ✅
* Check the name against https://repology.org/ ⚠️
Search for a lib with the same name, we recommend changing the name
* Check the name against Bing/Google. (If the first results aren't C++ related, try "name C++")
⚠️https://github.com/MikePopoloski/slang
* Check the source code for optional `find_package`s
No
* Check that the versioning scheme matches what upstream says ✅
* Check that the license in vcpkg.json matches what upstream says ✅
* Check that the license installed by the port matches what upstream says ✅
* Check that the source code comes from the upstream project's authoritative source. ✅
* Check that the generated usage is accurate. No
```

**Comment by djohansson** [comment]
@JonLiu1993 what is the next step? Should I rename the package?

**Comment by JonLiu1993** [comment]
> @JonLiu1993 what is the next step? Should I rename the package?



> @JonLiu1993 what is the next step? Should I rename the package?

Yes, rename library.

**Comment by JonLiu1993** [comment]
@djohansson ，[This](https://repology.org/projects/?search=shader-slang ) library seems to have the same name as the new one, and shader-libslang doesn't seem to have the same name. What do you think of this?

**Comment by djohansson** [comment]
@JonLiu1993 the library you pointed to is actually the same as this one from what I can tell. So shader-slang should be a valid name then?

**Comment by JonLiu1993** [comment]
> shader-slang

shader-slang, I think it's good.

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/shader-slang/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,83 @@
+set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
```
Comment (BillyONeal) [inline ports/shader-slang/portfile.cmake:1]: Line 72 installs headers, so why did you need this policy?

**Review by BillyONeal — CHANGES_REQUESTED** [review]
Couple notes:

* Normally duplicating release libs into the debug tree would be bad, because the CRT wouldn't match. However, this presumably passed our post build check that looks for that.
* LLVM is vendored, but only in binary forms. I think there is likely a concern on Linux over this, but maybe not.
* Several other libraries that are in vcpkg are vendored as well, as documented here: https://github.com/shader-slang/slang#license

The vendoring is a bit scary but *probably* OK given that this thing is being deployed in binary forms.

This "request changes" review is mainly about why `VCPKG_POLICY_EMPTY_INCLUDE_FOLDER` is being turned on.

**Comment by BillyONeal** [comment]
> New port checklist:

Thanks for checking these out for me @JonLiu1993 :)

**Comment by BillyONeal** [comment]
Hmmm scratch all my comments about vendoring; it looks like those parts are used only in the tool binaries, which makes it conclusively fine. Just the question about empty include folder...

File: ports/shader-slang/portfile.cmake (hunk 2 above)
Comment (djohansson) [reply ports/shader-slang/portfile.cmake]: It was added by mistake, copy-pasting from other ports. I have removed it now.

**Review by BillyONeal — DISMISSED** [review]

**Comment by BillyONeal** [comment]
Thanks!
